operational_analysis/toolkits/reanalysis_downloading.py

```diff
@@ -0,0 +1,369 @@
+        var_dict (:obj:`dict`, optional): Optional dictionary mapping the desired reanalysis variable names from
+            PlanetOS to custom variable names which will be used in the datarame that is returned. Note that if the
+            argument var_names is undefined, the downloaded default variables will be converted to standard OpenOA
+            variable names. Defaults to None.
+        calc_derived_vars (:obj:`bool`, optional): Boolean that specifies whether wind speed, wind direction, and air
+            density are computed from the downloaded reanalysis variables. Note that this requires the downloaded
+            eastward wind speed, northward wind speed, temperature, and surface pressure to be renamed to the standard
+            variable names "u_ms", "v_ms", "temperature_K", and "surf_pres_Pa". Defaults to False.
+        save_pathname (:obj:`string`, optional): The path where the downloaded reanalysis data will be saved (if
+            defined). Defaults to None.
+        save_filename (:obj:`string`, optional): The file name used to save the downloaded reanalysis data (if
+            defined). Defaults to None.
+        apikey_file (:obj:`string`, optional): The combined path and file name where the PlanetOS API key is saved. If
+            undefined, will assume there is a file called "APIKEY" saved in the operational_analysis.toolkits
+            directory. Defaults to None.
+
+    Returns:
+        :obj:`pandas.DataFrame`: A dataframe containing time series of the requested reanalysis variables
+    """
+
+    # Import PlanetOS apikey
+    if apikey_file is None:
+        apikey_file = Path(__file__).parent.resolve() / "APIKEY"
+    apikey = open(apikey_file).read().strip()
+
+    # Get start and end dates of dataset
+    start_date_ds, end_date_ds = get_dataset_start_end_dates_planetos(dataset, apikey)
+
+    # Depending on the start date, end date, and number of years specified, calculate the appropriate start and end
+    # dates
+    start_date, end_date = _get_start_end_dates_planetos(
+        start_date, end_date, num_years, start_date_ds, end_date_ds
+    )
+
+    # Determine number of data points
+    date_diff = end_date - start_date
+    count = int(24 * date_diff.days + date_diff.seconds / 3600)
+
+    # Get default variable names and dictionary if not defined
+    if var_names is None:
+        var_dict = _get_default_var_dicts_planetos(dataset)
+        var_names = list(var_dict.keys())
+
+    # Download data from PlanetOS
+    url = "http://api.planetos.com/v1/datasets/" + dataset_names[dataset] + "/point?"
+
+    kwgs = {
+        "apikey": apikey,
+        "count": count,
+        "lon": lon,
+        "lat": lat,
+        "vars": ", ".join(var_names),
+        "time_start": start_date.strftime("%Y-%m-%dT%H:%M:%S"),
+        "time_end": end_date.strftime("%Y-%m-%dT%H:%M:%S"),
+    }
+
+    r = requests.get(url, params=kwgs)
```

Comment:
Should we be allowing for the user to add in additional keyword arguments here by passing `**kwargs` into the function definition, or is that too niche of a use case?

Reply:
That's a good idea. I added a `**kwargs` argument that then gets appended to the standard kwargs dictionary used in the API call.